planner/core/physical_plan_test.go

```diff
@@ -851,7 +851,7 @@ func (s *testPlanSuite) TestDAGPlanBuilderAgg(c *C) {
 		},
 		{
 			sql:  "select (select count(1) k from t s where s.a = t.a having k != 0) from t",
-			best: "LeftHashJoin{TableReader(Table(t))->TableReader(Table(t)->StreamAgg)->StreamAgg->Sel([ne(k, 0)])}(test.t.a,s.a)->Projection->Projection",
+			best: "MergeLeftOuterJoin{TableReader(Table(t))->TableReader(Table(t))->Projection}(test.t.a,s.a)->Projection->Projection",
```

Comment:
First `decorrelate`, it become `select k from t left join (select count(1) k from t s group by a) on s.a=t.a and k != 0`.
Then `aggregate eliminate`, it become `select k from t left join (select if(isnull(1), 0, 1) k from t s) on s.a=t.a and k != 0`. Since if(isnull(1), 0, 1) is always 1. So it simplified as `select k from t left join (select 1 k from t s) on s.a=t.a and k != 0`.
Finally `predicate push down`, `k != 0` pushed down and converted to `1 != 0` which is always true.
So `Sel([ne(k, 0)])` is removed after `aggregate eliminate` is operated after `decorrelate`.